### PR TITLE
fix: Add version to the proxy cache to resolve concurrency issues

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #36989

### CODE REVIEW

File: internal/proxy/meta_cache.go [hunk 1]
```diff
@@ -368,18 +372,23 @@ func InitMetaCache(ctx context.Context, rootCoord types.RootCoordClient, queryCo
...
+		credMap:                map[string]*internalpb.CredentialInfo{},
+		shardMgr:               shardMgr,
+		privilegeInfos:         map[string]struct{}{},
+		userToRoles:            map[string]map[string]struct{}{},
+		collectionCacheVersion: make(map[string]uint64),
 	}, nil
 }
 
+func buildVersionKey(database, collectionName string) string {
+	return database + "-" + collectionName
```
Comment (czs007) [inline internal/proxy/meta_cache.go:389]: just use collectionID?

**Comment by xiaocai2333** [comment]
rerun go-sdk

**Comment by xiaocai2333** [comment]
rerun go-sdk

**Comment by czs007** [comment]
rerun go-sdk

**Comment by xiaocai2333** [comment]
rerun go-sdk

**Comment by xiaocai2333** [comment]
rerun ut

**Comment by xiaocai2333** [comment]
rerun ut

**Pull request merged**